packages/prop-house-backend/src/auction-watcher/auction-watcher.service.ts

```diff
@@ -0,0 +1,117 @@
+import { Injectable, Logger } from '@nestjs/common';
+import { EventEmitter2 } from '@nestjs/event-emitter';
+import { Cron } from '@nestjs/schedule';
+import { Auction } from 'src/auction/auction.entity';
+import { AuctionsService } from 'src/auction/auctions.service';
+import { AuctionClosedEvent } from 'src/auction/events/auction-closed.event';
+import { AuctionCreatedEvent } from 'src/auction/events/auction-created.event';
+import { AuctionOpenEvent } from 'src/auction/events/auction-open.event';
+import { AuctionProposalEndingSoonEvent } from 'src/auction/events/auction-proposal-end-soon.event';
+import { AuctionVotingEndingSoonEvent } from 'src/auction/events/auction-vote-end-soon.event';
+import { AuctionVotingEvent } from 'src/auction/events/auction-voting.event';
+
+@Injectable()
+export class AuctionWatcherService {
+  private readonly logger = new Logger(AuctionWatcherService.name);
+  constructor(
+    private readonly auctionService: AuctionsService,
+    private readonly events: EventEmitter2,
+  ) {}
+
+  @Cron('*/15 * * * * *')
+  async handleCron() {
+    this.logger.debug('Checking for Auction status changes');
+    const allAuctions = (await this.auctionService.findAllExtended()).filter(
+      (auction: Auction) => !auction.finalized(),
+    );
```

Comment:
Would suggest tightening the query to only fetch future/ongoing auctions, rather than fetching all auctions (including all props) and filtering in JS.

TypeORM operators in where clause should work -

```ts
votingEndTime: MoreThan(new Date()),
eventStatus: Not(['auctionClosed', null]),
```

Reply:
TypeORM's `Not` doesn't split those like described and we do want to see `null`'s since they've yet to be "discovered" for legacy rounds. The rest worked great though, good suggestion.

Reply:
Missing the `Never processed but was also never discovered` case when the voting end time is in the past then, no?

Haven't tested, but may be looking for `eventStatus: Not(In(['auctionClosed', null]))` instead.